src/App.js

```diff
@@ -0,0 +1,116 @@
+import React, { useState, useCallback, useRef } from 'react';
```

Comment:
This is structured really well!  I think some commentary throughout might've helped with legibility for code review - but still certainly readable!